src/utils/mod.rs

```diff
@@ -274,6 +308,117 @@ impl TpmsRsaParmsBuilder {
 /// Supported sizes for RSA key modulus
 pub const RSA_KEY_SIZES: [u16; 4] = [1024, 2048, 3072, 4096];
 
+/// Builder for `TPMS_ECC_PARMS` values.
+// Most of the field types are from bindgen which does not implement Debug on them.
+#[allow(missing_debug_implementations)]
+#[derive(Copy, Clone)]
+pub struct TpmsEccParmsBuilder {
+    /// Symmetric cipher to be used in conjuction with the key
+    pub symmetric: Option<Cipher>,
+    /// Asymmetric scheme to be used for key operations
+    pub scheme: AsymSchemeUnion,
+    /// Curve to be used with the key
+    pub curve: EllipticCurve,
+    /// Flag indicating whether the key shall be used for signing
+    pub for_signing: bool,
+    /// Flag indicating whether the key shall be used for decryption
+    pub for_decryption: bool,
+    /// Flag indicating whether the key is restricted
+    pub restricted: bool,
+}
+
+impl TpmsEccParmsBuilder {
+    /// Create parameters for a restricted decryption key (i.e. a storage key)
+    pub fn new_restricted_decryption_key(symmetric: Cipher, curve: EllipticCurve) -> Self {
```

Comment:
What is the justification for the decryption one to be restricted and the signing one to be unrestricted?

Reply:
Restricted decryption keys are essentially Parent keys - they could be either used for derivation or storage of other object underneath it. The ECC one isn't used like that, so I could remove this

Reply:
Ahhh got it thanks!